docarray/array/storage/base/getsetdel.py

```diff
@@ -260,7 +262,7 @@ def _set_doc_attr_by_id(self, _id: str, attr: str, value: Any):
 
         d = self._get_doc_by_id(_id)
         if hasattr(d, attr):
-            setattr(d, attr, value)
+            setattr(d, attr, detach_tensor_if_present(value))
```

Comment:
I would prefer to use if statement here, I believe it would be much safer and more efficient.
```
if attr in ['embedding', 'tensor']:
    value = detach_tensor(value)
    ...
```